test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -2061,6 +2067,156 @@ public async Task RestoreCommand_CentralVersion_AssetsFile_VerifyProjectsReferen
             }
         }
 
+        /// <summary>
+        /// Verifies an error is logged when a user attempts to specify a VersionOverride but the feature is disabled and that restore succeeds if the feature is enabled.
+        /// </summary>
+        /// <returns></returns>
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task RestoreCommand_CentralVersion_ErrorWhenVersionOverrideUsedButIsDisabled(bool isCentralPackageVersionOverrideDisabled)
+        {
+            const string projectName = "TestProject";
+
+            const string packageName = "PackageA";
+
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                string projectPath = Path.Combine(pathContext.SolutionRoot, projectName);
+                string outputPath = Path.Combine(projectPath, "obj");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, PackageSaveMode.Defaultv3, new SimpleTestPackageContext(packageName, "1.0.0"));
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, PackageSaveMode.Defaultv3, new SimpleTestPackageContext(packageName, "2.0.0"));
+
+                var packageRefDependecyFoo = new LibraryDependency()
+                {
+                    LibraryRange = new LibraryRange(packageName, versionRange: null, typeConstraint: LibraryDependencyTarget.Package),
+                    VersionOverride = new VersionRange(NuGetVersion.Parse("2.0.0"))
+                };
+
+                var packageVersion = new CentralPackageVersion(packageName, VersionRange.Parse("1.0.0"));
```

Comment:
You might have an easier time with these if you get them through the GetPackageSpec. 

You'd reduce the the potential inconsistencies in the package spec.s

Reply:
I copy/pasted this pattern from existing tests.  I'm going to leave it as-is for now, maybe I can clean up these tests later